hypothesis-python/tests/cover/test_example.py

```diff
@@ -66,3 +73,17 @@ def test_example_inside_find():
 @fails_with(HypothesisException)
 def test_example_inside_strategy():
     st.booleans().map(lambda x: st.integers().example()).example()
+
+
+def test_non_interactive_example_emits_warning():
+    # We have this warning disabled for most of our tests, because self-testing
+    # Hypothesis means `.example()` can be a good idea when it never is for users.
+    with warnings.catch_warnings():
+        warnings.simplefilter("always")
+        with pytest.warns(NonInteractiveExampleWarning):
+            st.text().example()
+
+
+def test_interactive_example_does_not_emit_warning():
+    code = "__import__('hypothesis').strategies.none().example()"
+    subprocess.check_call([sys.executable, "-c", code])
```

Comment:
I don’t think you have `sys.ps1` when running with `python -c` – I just tried it, and I get an AttributeError:

```console
$ python3.6 -c 'import sys; sys.ps1'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: module 'sys' has no attribute 'ps1'
```

Also, if this code did drop a warning, would the test catch it? I think we need `[sys.executable, '-Werr', …` if you want this `check_call` to throw on a warning; otherwise it gets logged to no effect.

Reply:
😭 

You're entirely right, of course... thanks for pointing it out though ❤️